File: wellness-mentor-service/src/main/java/com/stackroute/wellnessmentorservice/model/Mentor.java
package com.stackroute.wellnessmentorservice.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;


@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Document(collection = "mentor")
public class Mentor {
    @Id
    private String emailId;
    private String name;
    private String expertize;
    private int rating;
    private String imagePath;
    private String about;
    private String education;
    private String certification;
    private String training;
    private String experience;
    private String language;
    private String specialities;
    private AppointmentSlot[] appointmentSlots;
    private BookedAppointment[] bookedAppointments;
}

File: wholesome-care-frontend/src/app/mentor-profile/Mentor.ts
import { BookedAppointment } from "./BookedAppointment";
import { AppointmentSlot } from "./AppointmentSlot";
export class Mentor{
    emailId: string;
    name: string;
    expertize: string;
    rating: number;
    imagePath: String;
    about: String;
    education: String;
    certification: String;
    training: String;
    experience: String;
    language:String;
    specialities: String;
    appointmentSlots: AppointmentSlot[];
    bookedAppointments:BookedAppointment[];

    constructor(){
        this.emailId='';
        this.name='';
        this.expertize='';
        this.rating=0;
        this.imagePath='';
        this.about='';
        this.education='';
        this.certification='';
        this.training='';
        this.experience='';
        this.language='';
        this.specialities='';
        this.appointmentSlots = [new AppointmentSlot()];
        this.bookedAppointments=null;

    }

}